hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/action/rollback/TestCopyOnWriteRollbackActionExecutor.java

```diff
@@ -328,4 +340,213 @@ private void performRollbackAndValidate(boolean isUsingMarkers, HoodieWriteConfi
 
     assertFalse(WriteMarkersFactory.get(cfg.getMarkersType(), table, commitInstant.getTimestamp()).doesMarkerDirExist());
   }
+
+  /**
+   * This method tests rollback of completed ingestion commits and replacecommit inflight files
+   * when there is another replacecommit with greater timestamp already present in the timeline.
+   */
+  @Test
+  public void testRollbackWhenReplaceCommitIsPresent() throws Exception {
```

Comment:
can you move these into separate PRs. will help landing the PRs at diff pace. also, the PR description will be succinct. its fine for this PR. but in future